### PR TITLE
Opinionated language packages?

### PR DESCRIPTION
**Pull request by lee-dohm**
[Rendered](https://github.com/atom/design-decisions/blob/opinionated-language-packages/opinionated-language-packages.md)

### CODE REVIEW

**Comment by joshaber** [comment]
I'm inclined to say we shouldn't be opinionated except where syntax itself dictates.

Text editors are personal for programmers which is why we love Atom.

**Comment by lee-dohm** [comment]
So would you support reverting the setting in language-python that defaults to four spaces?

**Comment by joshaber** [comment]
I think so, but I'm really curious to hear what others think.

**Comment by Alhadis** [comment]
Just out of curiosity, how are tabs baneful in Python? They seem like the most logical choice of indentation style for a whitespace-significant language.

**Comment by lee-dohm** [comment]
@Alhadis You'd have to ask Guido Van Rossum and the other authors of the [PEP8](https://www.python.org/dev/peps/pep-0008/). The question here isn't "tabs vs spaces" in any case.

**Comment by lee-dohm** [comment]
I agree that we shouldn't be opinionated, we should only support syntactic requirements. Personally, I feel like Python is a special exception because it is:
1. A style guide written by the author of the language
2. Intended to be followed in all cases even if not enforced by the language itself

Similarly, I would support updating the [language-go package](https://atom.io/packages/language-go) to have defaults in line with the [gofmt tool](https://golang.org/cmd/gofmt/) (even though the gofmt tool itself makes such an effort unnecessary).

**Comment by Alhadis** [comment]
I've written Python using hard tabs before. Unlike YAML, spaces aren't a syntactic requirement of Python. Therefore, there's no logical reason a user MUST use spaces as their indent style when authoring  Python code. Having the editor unexpectedly change indentation style runs counter to user expectations.

Exceptions shouldn't be made on a "users should" basis, because that introduces a slippery slope. Look at PHP's PEP guidelines. Note how they also mandate the use of 4-spaces instead of tabs. Since that's what the community recommendation is, should Atom be obliged to honour that, too? What about Rust, whose maintainers WANT you to use spaces instead of tabs?  All of these pose questions that are difficult to answer short of admittedly arbitrary logic like "that's what the community's agreed on".

In other words, it's a Pandora's Box. Best leave it closed.

**Comment by lee-dohm** [comment]
@Alhadis Thank you very much for your feedback. We have heard from you on this multiple times and we know your stance. It is time to allow the Atom team to discuss what we are comfortable with supporting and why.
